web/src/app/admin/admin-message-logs/useMessageLogs.ts

```diff
@@ -0,0 +1,141 @@
+import _ from 'lodash'
+import React, { useLayoutEffect, useEffect, useRef, useState } from 'react'
+import { gql, useClient } from 'urql'
+import { Alert, DebugMessage, MessageLogSearchOptions } from '../../../schema'
+
+export const logsQuery = gql`
+  query messageLogsQuery($input: MessageLogSearchOptions) {
+    messageLogs(input: $input) {
+      nodes {
+        id
+        createdAt
+        updatedAt
+        type
+        status
+        userID
+        userName
+        source
+        destination
+        serviceID
+        serviceName
+        alertID
+        providerID
+      }
+      pageInfo {
+        hasNextPage
+        endCursor
+      }
+    }
+  }
+`
+
+const QUERY_LIMIT = 30
+
+export type MessageLogData = {
+  logs: DebugMessage[]
+  loading: boolean
+  error: Error | undefined
+}
+
+// useMessageLogs will fetch up message logs up to the QUERY_LIMIT set above
+// using the standard pagination endpoint.
+// modeled after useAlerts.ts
+export function useMessageLogs(
+  options: MessageLogSearchOptions,
+  depKey: string,
+  pause?: boolean,
+): MessageLogData {
+  const { createdBefore, createdAfter, omit, search } = options
+  const [logs, setLogs] = useState<DebugMessage[]>([])
+  const [loading, setLoading] = useState(true)
+  const [error, setError] = useState<Error | undefined>()
+  const key = useRef(depKey)
+  key.current = depKey
+
+  useEffect(() => {
+    return () => {
+      // cancel on unmount
+      key.current = ''
+    }
+  }, [])
+
+  const client = useClient()
+  const fetch = React.useCallback(async () => {
+    setLogs([])
+    setLoading(true)
+    setError(undefined)
+    if (pause) {
+      return
+    }
+    async function fetchLogs(
+      cursor: string,
+    ): Promise<[Alert[], boolean, string, Error | undefined]> {
+      const q = await client
+        .query(logsQuery, {
+          input: {
+            first: QUERY_LIMIT,
+            after: cursor,
+            createdBefore,
+            createdAfter,
+            omit,
+            search,
+          },
+        })
+        .toPromise()
+
+      if (q.error) {
+        return [[], false, '', q.error]
+      }
+
+      return [
+        q.data.messageLogs.nodes,
+        q.data.messageLogs.pageInfo.hasNextPage,
+        q.data.messageLogs.pageInfo.endCursor,
+        undefined,
+      ]
+    }
+
+    const throttledSetLogs = _.throttle(
+      (logs, loading) => {
+        setLogs(logs) // todo: additional sorting needed?
```

Comment:
It may be helpful to sort by createdAt here.

Reply:
The idea is that it should always come back sorted from the backend, but I'll revisit this to be sure so we don't put any extra load on the front end since it's already doing it's best to get everything loaded

Reply:
Sounds good 👍 I'll make sure we sort on the backend to avoid any extra strain on the browser while it's loading in data.